File: JavaScript_Basic_04_2023_SolutionsOfProblems/03.Ex_PB-JS-Nested-Conditional-Statements/P09_SkiTrip.js
function skiTrip(input) {
    let dayStay = Number(input[0]);
    let roomType = input[1];
    let ratingType = input[2];

    let holidayCost = 0.0;

    switch (roomType) {
        case "room for one person":
            holidayCost = (dayStay - 1) * 18;
            break;
        case "apartment":
            holidayCost = (dayStay - 1) * 25;
            if (dayStay > 15) {
                holidayCost = holidayCost * 0.5;
            } else if (dayStay >= 10) {
                holidayCost = holidayCost * 0.65;
            } else {
                holidayCost = holidayCost * 0.7;
            }
            break;
        case "president apartment":
            holidayCost = (dayStay - 1) * 35;
            if (dayStay > 15) {
                holidayCost = holidayCost * 0.8;
            } else if (dayStay >= 10) {
                holidayCost = holidayCost * 0.85;
            } else {
                holidayCost = holidayCost * 0.9;
            }
            break;
    }
    if (ratingType === "positive") {
        holidayCost = holidayCost * 1.25;
    } else {
        holidayCost = holidayCost * 0.9;
    }
    console.log(holidayCost.toFixed(2));
}

skiTrip(["2",
"apartment",
"positive"])

File: SoftUni_JAVA_BASIC_Jan_2023_SolutionsOfProblems/P03Ex_ConditionalStatements_Advanced/P09_SkiTrip.java
package P03Ex_ConditionalStatements_Advanced;

import java.util.Scanner;

public class P09_SkiTrip {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int daysStay = Integer.parseInt(scanner.nextLine());
        String roomType = scanner.nextLine();
        String review = scanner.nextLine();

        double price = 0;

        switch (roomType) {
            case "room for one person":
                price = (daysStay - 1) * 18;
                break;
            case "apartment":
                price = (daysStay - 1) * 25;
                if (daysStay > 15) {
                    price = price * 0.5;
                } else if (daysStay > 10) {
                    price = price * 0.65;
                } else {
                    price = price * 0.7;
                }
                break;
            case "president apartment":
                price = (daysStay - 1) * 35;
                if (daysStay > 15) {
                    price = price * 0.8;
                } else if (daysStay > 10) {
                    price = price * 0.85;
                } else {
                    price = price * 0.9;
                }
                break;
        }

        if (review.equals("positive")) {
            price = price * 1.25;
        } else {
            price = price * 0.9;
        }

        System.out.printf("%.2f", price);
    }
}
